### PR TITLE
system transaction

### PR DESCRIPTION
**Pull request by zjma**
### Description

This PR introduces a new transaction type `SystemTransaction` and a block type `ProposalExt` that supports it.

### TODOs
- [ ] Indexer-side updates.

### CODE REVIEW

File: ecosystem/indexer-grpc/indexer-grpc-fullnode/src/convert.rs [hunk 1]
```diff
@@ -725,6 +725,9 @@ pub fn convert_transaction(
             transaction::transaction::TransactionType::StateCheckpoint
         },
         Transaction::PendingTransaction(_) => panic!("PendingTransaction is not supported"),
+        Transaction::SystemTransaction(_) => {
+            todo!()
```
Comment (zjma) [inline ecosystem/indexer-grpc/indexer-grpc-fullnode/src/convert.rs:729]: cc @bowenyang007

File: ecosystem/indexer-grpc/indexer-grpc-fullnode/src/convert.rs [hunk 2]
```diff
@@ -775,6 +778,9 @@ pub fn convert_transaction(
             )
         },
         Transaction::PendingTransaction(_) => panic!("PendingTransaction not supported"),
+        Transaction::SystemTransaction(_) => {
+            todo!()
```
Comment (zjma) [inline ecosystem/indexer-grpc/indexer-grpc-fullnode/src/convert.rs:782]: cc @bowenyang007

File: crates/indexer/src/models/system_transactions.rs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+#[derive(
+    Associations, Clone, Debug, Deserialize, FieldCount, Identifiable, Insertable, Serialize,
+)]
+#[diesel(belongs_to(Transaction, foreign_key = version))]
+#[diesel(primary_key(version))]
+#[diesel(table_name = system_transactions)]
+pub struct SystemTransaction {
+    pub version: i64,
+    pub block_height: i64,
+    pub txn_serialized: Vec<u8>,
```
Comment (grao1991) [inline crates/indexer/src/models/system_transactions.rs:23]: Do we really need to store the raw txn?

File: api/types/src/transaction.rs [hunk 4]
```diff
@@ -523,6 +555,16 @@ pub struct BlockMetadataTransaction {
...
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize, Object)]
+pub struct SystemTransaction {
+    #[serde(flatten)]
+    #[oai(flatten)]
+    pub info: TransactionInfo,
+    pub events: Vec<Event>,
+    pub timestamp: U64,
+    pub txn_serialized: Vec<u8>,
```
Comment (grao1991) [inline api/types/src/transaction.rs:565]: Should this be an enum?

File: crates/indexer/src/models/system_transactions.rs (hunk 3 above)
Comment (grao1991) [reply crates/indexer/src/models/system_transactions.rs:23]: @bowenyang007

Comment (zjma) [reply crates/indexer/src/models/system_transactions.rs:23]: All the indexer changes in this PR so far were made just to get rid of compile errors...
It doesn't have to be like this :)

File: api/types/src/transaction.rs (hunk 4 above)
Comment (zjma) [reply api/types/src/transaction.rs:565]: probably. But let's also expect this enum will have new variant every ~3 months. Will that hurt?

Comment (grao1991) [reply api/types/src/transaction.rs:565]: I don't see any problem here.

File: crates/indexer/src/models/transactions.rs [hunk 5]
```diff
@@ -219,6 +220,33 @@ impl Transaction {
...
+                        SystemTransaction::from_transaction(sys_txn, block_height),
+                    )),
+                    EventModel::from_events(
+                        &sys_txn.events,
+                        sys_txn.info.version.0 as i64,
+                        block_height,
+                    ),
+                    wsc,
+                    wsc_detail,
+                )
```
Comment (bowenyang007) [inline crates/indexer/src/models/transactions.rs:248]: we don't need to add anything here. Just add a default that returns null is sufficient. I can handle this.

File: crates/indexer/src/models/system_transactions.rs (hunk 3 above)
Comment (bowenyang007) [reply crates/indexer/src/models/system_transactions.rs:23]: Yea I can clean everything in this crate up. This is all unnecessary b/c we're no longer adding new features to this part of the indexer.

File: api/types/src/transaction.rs (hunk 4 above)
Comment (bowenyang007) [inline api/types/src/transaction.rs:565]: What is this txn_serialized field? Could you paste the output of an example transaction here? E.g. something like this: https://devnet.aptoslabs.com/v1/transactions/by_version/25100

**Pull request closed without merging**